staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go

```diff
@@ -266,12 +266,7 @@ func setVisitor(visitorConfig *driver.VisitorConfig, twin *common.Twin, dev *dri
 		return nil
 	}
 	klog.V(2).Infof("Convert type: %s, value: %s ", twin.Property.PProperty.DataType, twin.ObservedDesired.Value)
-	value, err := common.Convert(twin.Property.PProperty.DataType, twin.ObservedDesired.Value)
-	if err != nil {
-		klog.Errorf("Failed to convert value as %s : %v", twin.Property.PProperty.DataType, err)
-		return err
-	}
-	err = dev.CustomizedClient.SetDeviceData(value, visitorConfig)
+	err := dev.CustomizedClient.SetDeviceData(twin.ObservedDesired.Value, visitorConfig)
```

Comment:
The purpose of convert here is to convert string type `value` into other types, such as int, etc. This type is consistent with the user-defined device property type.

Reply:
The data type of device property is universal, so we define it in `devicemodel` https://github.com/kubeedge/kubeedge/blob/master/staging/src/github.com/kubeedge/api/apis/devices/v1beta1/device_model_types.go#L41 . `deviceinstance` needs to obtain the data type of the device property from` devicemodel` when building https://github.com/kubeedge/kubeedge/blob/master/staging/src/github.com/kubeedge/mapper-framework/pkg/util/parse/grpc.go#L280-L303

Reply:
My field on the model is int, but if the expected value is not set by default, it will be "", which will cause the mapper to fail to start by default. This is equivalent to all my int type fields having to be set to a number by default.

Reply:
is this expected behavior because user specifies that empty string value into the int type, that sounds like syntax error? this PR returns to the application with empty string without any error which appears to be the success of conversion (from int in your particular case), but internally it does not succeed any conversion at all?

Reply:
I also agree that we need to think about whether an empty string is an illegal input or a zero value according to the scenario.

Reply:
+1, and I suggest that you can add null value verification to the place where this convert function is called, that is, `setvisitor`. If the expected value is null, you can output the log and returned directly. There is no need to enter the `convert` function.

Reply:
> +1, and I suggest that you can add null value verification to the place where this convert function is called, that is, `setvisitor`. If the expected value is null, you can output the log and returned directly. There is no need to enter the `convert` function.+1，我建议你可以在调用这个convert函数的地方，即`setvisitor`添加空值验证。如果期望值为null，则可以输出日志并直接返回。无需输入`convert`函数。

I think this suggestion is good.